tests/support/mod.rs

```diff
@@ -150,17 +153,19 @@ impl RlsHandle {
         let mut stdout_len = 0;
         loop {
             let stdout = self.stdout();
-            if stdout.0.len() != stdout_len {
+            if stdout.out.len() != stdout_len {
                 if stdout_predicate(&stdout) {
                     break stdout;
                 }
-                stdout_len = stdout.0.len();
+                stdout_len = stdout.out.len();
             }
 
             assert!(
-                start.elapsed() < timeout,
-                "Timed out waiting {:?} for predicate",
-                timeout
+                start.elapsed().min(stdout.last_write.elapsed()) < timeout
+                    && start.elapsed() < timeout * 10,
```

Comment:
`start.elapsed() < timeout * 10` here is just a little defence against rls drip feeding stdout for some reason in the future.